src/lib/peers/MulticastBroadcaster.ts

```diff
@@ -58,118 +46,77 @@ class MulticastBroadcaster extends EventEmitter {
       (() => {
         this.socket.addMembership(UDP_MULTICAST_ADDR);
         const address = this.socket.address();
+
+        // Start the broadcasting process
+        this.startBroadcasting();
       }).bind(this),
     );
-
-    // Handle messages
-    this.socket.on('message', this.handleHandshakeMessages.bind(this));
-
-    // Start the query process
-    this.queryInterval = this.queryLAN();
   }
 
-  /**
-   * Request a peer contact for the multicast peer discovery to check for
-   * @param publicKey Public key of the desired peer
-   */
-  async requestPeerContact(publicKey: string) {
-    const pubKeyBuf = Buffer.from(publicKey);
-    const randomMessage = crypto.randomBytes(16);
-    // Encrypt message
-    const encryptedPeerPubKey = await this.keyManager.encryptData(pubKeyBuf, pubKeyBuf);
-    const encryptedRandomMessage = await this.keyManager.encryptData(randomMessage, pubKeyBuf);
-    const encryptedLocalPubKey = await this.keyManager.encryptData(
-      Buffer.from(this.keyManager.getPublicKey()),
-      pubKeyBuf,
-    );
+  startListening() {
```

Comment:
This is related to stealth mode?

Also can we make these messages protobuf messages?

Reply:
yes and yes